### ISSUE TITLE
Problem with getting gradient from model parameters 

### ISSUE THREAD

**Issue by MetalistForever** [ClippedStochasticMethods/ZO-clipped-SSTM#1]
## Problem
param.grad in `cal_one_repeat` method returns `None`

## Description
I ran the code without out of the box, and this bug in the first training loop appeared:
![image](https://github.com/ClippedStochasticMethods/ZO-clipped-SSTM/assets/56347723/cd1d2b51-ac2f-4c6d-baea-64c1ad4b230a)

However, this issue looks like abnormal, because everything should work fine (all cell were ran one after one in the original notebook). Can you clarify which requirements were used when running the code, please?

## Used setting:
torch == 2.0.0+cu118
numpy==1.26.4
scipy==1.12.0
